core/ecschema-editing/src/test/Merging/CustomAttributeMerger.test.ts

```diff
@@ -0,0 +1,309 @@
+            ],
+          },
+          TestEntity: {
+            schemaItemType: "EntityClass",
+            properties: [
+              {
+                name: "DoubleProp",
+                type: "PrimitiveProperty",
+                typeName: "double",
+              },
+              {
+                name: "DateTimeProp",
+                type: "PrimitiveProperty",
+                typeName: "dateTime",
+              },
+            ],
+          },
+        },
+      }, targetContext);
+
+      const merger = new SchemaMerger();
+      const mergedSchema = await merger.merge(targetSchema, sourceSchema);
+      const mergedEntity = await mergedSchema.getItem<EntityClass>("TestEntity");
+      expect(mergedEntity!.toJSON().properties).deep.eq(
+        [
+          {
+            name: "DoubleProp",
+            type: "PrimitiveProperty",
+            typeName: "double",
+            customAttributes: [
+              {
+                LongProp: 1.999,
+                className: "TargetSchema.TestCA",
+              },
+            ],
+          },
+          {
+            name: "DateTimeProp",
+            type: "PrimitiveProperty",
+            typeName: "dateTime",
+            customAttributes: [
+              {
+                IntProp: 25,
+                StringPrimitiveArrayProp: [
+                  "PropertyCustomAttribute",
+                ],
+                className: "TestSchema.TestCA",
+              },
+            ],
+          },
+        ],
+      );
+    });
+
+    it("should merge missing schema custom attributes", async () => {
+      const sourceSchema = await Schema.fromJson({
+        ...sourceJson,
+        references: [
+          {
+            name: "TestSchema",
+            version: "01.00.15",
+          },
+        ],
+        customAttributes: [
+          {
+            className: "SourceSchema.TestCA",
+          },
+          {
+            StringPrimitiveArrayProp: [
+              "SchemaCustomAttribute",
+            ],
+            className: "TestSchema.TestCA",
+          },
+        ],
+        items: {
+          TestCA: {
+            schemaItemType: "CustomAttributeClass",
+            appliesTo: "Schema",
+          },
+        },
+      }, sourceContext);
+
+      const targetSchema = await Schema.fromJson({
+        ...targetJson,
+        items: {
+          TestCA: {
+            schemaItemType: "CustomAttributeClass",
+            appliesTo: "Schema",
+          },
+        },
+      }, targetContext);
+
+      const merger = new SchemaMerger();
+      const mergedSchema = await merger.merge(targetSchema, sourceSchema);
+      expect(mergedSchema.toJSON().customAttributes).deep.eq(
+        [
+          {
+            className: "TargetSchema.TestCA",
+          },
+          {
+            StringPrimitiveArrayProp: [
+              "SchemaCustomAttribute",
+            ],
+            className: "TestSchema.TestCA",
+          },
+        ],
+      );
+    });
```

Comment:
you need a  test where the CA is defined in a schema referenced by the source but not by the target.